vonk/plugins/plugins_order.rst

```diff
@@ -0,0 +1,86 @@
+
+Also, you probably want to set your response on the ``IVonkContext.Response`` and not directly on the ``HttpContext.Response``. Then, you will need the :ref:`VonkToHttp <vonk_plugins_vonktohttp>` plugin (order: 1120) to translate the ``IVonkContext`` back to the ``HttpContext``. 
+
+So in general the minimum order is higher than 1120. 
+
+If you want your middleware to act on all the entries in a Batch or Transaction, you need to choose an order higher than that of the :ref:`Transaction <vonk_plugins_transaction>` plugin, which is 3120.
+
+.. _vonk_plugins_order_collisions:
+
+Order collisions
+----------------
+
+If two plugins define the same order, it is not defined in what order the plugins will be put in the pipeline. As long as those plugins act on disjoint sets of requests that may not be a problem. But it is recommended to avoid this by checking the orders in use already. 
+
+.. _vonk_plugins_order_prepost:
+
+Handlers and pre- and posthandlers
+----------------------------------
+
+In Vonk you can define different types of middleware:
+
+* Handler - acts on requests of a certain type, provides the response to it and ends the pipeline.
+* Prehandler - acts on requests of certain type(s), may modify the request and sends the request further down the pipeline.
+* Posthandler - lets the request pass by to be handled further down the pipeline. When the response passes on the way back, it acts on requests or responses of certain type(s), and may modify the response.
+
+This is explained in the `session on Plugins <https://www.youtube.com/watch?v=odYaOM19XXc>`_ from `DevDays 2018 <https://www.devdays.com/events/devdays-europe-2018/>`_ at 17:05.
+
+What type of middleware you want your service to be is defined by your use of one of the ``*Handle...`` methods from the :ref:`vonk_vonkappbuilder` or the :ref:`vonk_appbuilder_extensions`. 
+
+Prehandler
+^^^^^^^^^^
+
+A Prehandler needs to act *before* the actual handler will provide a response. So the order of it must be lower than any Handler that may handle the requests that this Prehandler is interested in.
+
+So if you want a Prehandler to intercept all create interactions, you should choose an order lower than that of the :ref:`Create <vonk_plugins_create>` plugin, which is 4420. 
+
+An example of this is the :ref:`Prevalidation <vonk_plugins_prevalidation>` plugin. It needs to validate all resources that get handled by the Create, Update, Conditional Create and Conditional Update plugins. Of these, Create has the lowest order: 4420. So it must be below 4420. But it also needs to act on each resource in a :ref:`Batch <vonk_plugins_batch>` or :ref:`Transaction <vonk_plugins_transaction>`, so it must be higher than these two, which means higher than 3120. So Prevalidation has order 4320.
```

Comment:
I would change the last sentence to something like: "So this is why we have chosen 4320 as order for Prevalidation"